### DISCUSSION TITLE
Download Artifacts Missing Files

### DISCUSSION THREAD

**Discussion by connorff** [community/community#26755] [Actions]
<p>I’m using the upload/download artifacts actions to reuse build outputs between jobs on the <code>split-install-jobs</code> branch of this <a href="https://github.com/connorff/flowistry/tree/split-install-jobs" rel="noopener nofollow ugc">repo</a>.</p>
<p>Here’s the job for creating the build outputs:</p>
<pre><code class="lang-auto">build:
    runs-on: ${{ matrix.os }}

    strategy:
      fail-fast: false
      matrix:
        os: [ubuntu-latest]

    steps:
      - uses: actions/checkout@v2
      - run: cargo install --path .
      - uses: actions/upload-artifact@v2
        with:
          name: flowistry-build
          path: |
            ~/.cargo/bin/cargo-flowistry
            ~/.cargo/bin/flowistry-driver
</code></pre>
<p>And this job attempts to test the project using the build (with added steps for debugging the missing files):</p>
<pre><code class="lang-auto">test:
    needs: build
    runs-on: ${{ matrix.os }}

    strategy:
      fail-fast: false
      matrix:
        os: [ubuntu-latest]

    steps:
      - uses: actions/checkout@v2
      - name: Retrieve Flowistry build
        uses: actions/download-artifact@v2
        with:
          name: flowistry-build
          path: |
            ~/.cargo/bin/cargo-flowistry
            ~/.cargo/bin/flowistry-driver
      - run: ls -a ~
      - run: ls ~/.cargo/bin/
      - run: ls -l ~/.cargo/bin/flowistry-driver
      - run: cargo flowistry forward_slice
      - run: cargo test
      - run: npm ci
        working-directory: ide
      - name: Run vscode tests
        run: xvfb-run -a npm test
        working-directory: ide
</code></pre>
<p>Here’s an <a href="https://github.com/connorff/flowistry/runs/4225483217" rel="noopener nofollow ugc">example run</a> of the workflows.</p>
<p>The artifacts from the action summary contains both <code>cargo-flowistry</code> <strong>and</strong> <code>flowistry-driver</code>:<br>
<img src="https://user-images.githubusercontent.com/40175891/181099768-54cc16d4-f5b0-4c7d-84a0-78bc286980ff.png" alt="image" data-base62-sha1="3jdrH2H3u1PouydUt2tOyGbRRCH" width="591" height="201"></p>
<p>The output for <code>ls ~/.cargo/bin/</code> in the workflow contains <code>cargo-flowistry</code>, but <strong>not</strong> <code>flowistry-driver</code>. And running <code>ls -l ~/.cargo/bin/flowistry-driver</code> confirms the file doesn’t exist:</p>
<pre><code class="lang-auto">ls: cannot access '/home/runner/.cargo/bin/flowistry-driver': No such file or directory
</code></pre>

**Comment by jsoref** [comment, marked as answer]
<p>As you sort of discovered here:</p><aside class="onebox githubcommit" data-onebox-src="https://github.com/connorff/flowistry/commit/91a5b41d62b118c2113ddd694546b137ab1e18d8">
  <header class="source">

      <a href="https://github.com/connorff/flowistry/commit/91a5b41d62b118c2113ddd694546b137ab1e18d8" target="_blank" rel="noopener nofollow ugc">github.com/connorff/flowistry</a>
  </header>

  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Commit">
    <svg width="60" height="60" class="github-icon" viewbox="0 0 14 16" aria-hidden="true"><path d="M10.86 7c-.45-1.72-2-3-3.86-3-1.86 0-3.41 1.28-3.86 3H0v2h3.14c.45 1.72 2 3 3.86 3 1.86 0 3.41-1.28 3.86-3H14V7h-3.14zM7 10.2c-1.22 0-2.2-.98-2.2-2.2 0-1.22.98-2.2 2.2-2.2 1.22 0 2.2.98 2.2 2.2 0 1.22-.98 2.2-2.2 2.2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/connorff/flowistry/commit/91a5b41d62b118c2113ddd694546b137ab1e18d8" target="_blank" rel="noopener nofollow ugc">`cp` flowistry build to cargo bin??</a>
    </h4>

    <div class="github-info">
      <div class="date">
        committed <span class="discourse-local-date" data-format="ll" data-date="2021-11-16" data-time="15:54:13" data-timezone="UTC">03:54PM - 16 Nov 21 UTC</span>
      </div>

      <div class="user">
        <a href="https://github.com/connorff" target="_blank" rel="noopener nofollow ugc">
          <img alt="connorff" src="https://user-images.githubusercontent.com/2119212/181099770-29438ad5-7770-400b-8508-8f8595a8ce92.jpeg" class="onebox-avatar-inline" width="20" height="20">
          connorff
        </a>
      </div>

      <div class="lines" title="changed 1 files with 1 additions and 3 deletions">
        <a href="https://github.com/connorff/flowistry/commit/91a5b41d62b118c2113ddd694546b137ab1e18d8" target="_blank" rel="noopener nofollow ugc">
          <span class="added">+1</span>
          <span class="removed">-3</span>
        </a>
      </div>
    </div>
  </div>
</div>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>the <code>path</code> field for <code>actions/download-artifact@v2</code> in:</p>
<pre><code class="lang-auto">      - name: Retrieve Flowistry build
        uses: actions/download-artifact@v2
        with:
          name: flowistry-build
          path: "this-is-a-single-path-not-a-list-of-paths"
</code></pre>
<p>… is a single directory (or sorts) into which artifacts are expanded.</p>
<aside class="onebox githubblob" data-onebox-src="https://github.com/actions/download-artifact/blob/b1985abdea713aec19e3253477b5a6472456a2fa/action.yml#L9">
  <header class="source">

      <a href="https://github.com/actions/download-artifact/blob/b1985abdea713aec19e3253477b5a6472456a2fa/action.yml#L9" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>

  <article class="onebox-body">
    <h4><a href="https://github.com/actions/download-artifact/blob/b1985abdea713aec19e3253477b5a6472456a2fa/action.yml#L9" target="_blank" rel="noopener nofollow ugc">actions/download-artifact/blob/b1985abdea713aec19e3253477b5a6472456a2fa/action.yml#L9</a></h4>



  <pre class="onebox">    <code class="lang-yml">
      <ol class="start lines" start="1" style="counter-reset: li-counter 0 ;">
          <li>name: 'Download a Build Artifact'</li>
          <li>description: 'Download a build artifact that was previously uploaded in the workflow by the upload-artifact action'</li>
          <li>author: 'GitHub'</li>
          <li>inputs: </li>
          <li>  name:</li>
          <li>    description: 'Artifact name'</li>
          <li>    required: false</li>
          <li>  path:</li>
          <li class="selected">    description: 'Destination path'</li>
          <li>    required: false</li>
          <li>runs:</li>
          <li>  using: 'node12'</li>
          <li>  main: 'dist/index.js'</li>
      </ol>
    </code>
  </pre>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>If you found the documentation confusing, I’d encourage you to file a bug against the repository.</p>

**Comment by connorff** [comment]
<p>Ah, that makes sense. The documentation seems good, more of an oversight on my part. Thanks!</p>
